python/gps/algorithm/algorithm_mdgps.py

```diff
@@ -96,39 +112,79 @@ def _update_policy(self):
             tgt_prc = np.concatenate((tgt_prc, prc))
             tgt_wt = np.concatenate((tgt_wt, wt))
             obs_data = np.concatenate((obs_data, samples.get_obs()))
-        self.policy_opt.update(obs_data, tgt_mu, tgt_prc, tgt_wt)
+        if 'fc_only_iterations' in self.policy_opt._hyperparams:
```

Comment:
This if statement seems a bit unnecessary here. Why can't we always pass in self.iteration_count, and put the if-statement in fc_only_iterations?